In [8]:
import tensorflow as tf

from numpy.random import RandomState

batch_size = 8

w1 = tf.Variable(tf.random_normal([2, 3], stddev=1, seed=1))
w2 = tf.Variable(tf.random_normal([3, 1], stddev=1, seed=1))

x = tf.placeholder(tf.float32, shape=(None, 2), name="x-input")
y_ = tf.placeholder(tf.float32, shape=(None, 1), name="y-input")

a = tf.nn.relu(tf.matmul(x, w1))
y = tf.nn.relu(tf.matmul(a, w2))

#a = tf.matmul(x, w1)
#y = tf.matmul(a, w2)
cross_entropy = -tf.reduce_mean(y_ * tf.log(tf.clip_by_value(y, 1e-10, 1.0)))
train_step = tf.train.AdamOptimizer(0.001).minimize(cross_entropy)
rdm = RandomState(1)
dataset_size = 128
X = rd.rand(dataset_size, 2)
Y = [[int(x1 * x1 * x1 + x2 * x2 * x2 < 100)] for (x1, x2) in X]

with tf.Session() as sess:
    init_op = tf.global_variable_initializer()
    sess.run(init_op)
    sess.run(w1)
    sess.run(w2)
    STEPS = 20000
    for i in range(STEPS):
        start = (i * batch_size) % dataset_size
        end = min(start + batch_size, dataset_size)
        sess.run(train_step, feed_dict={x: X[start:end], y_: Y[start:end]})
        
cross_entropy = -tf.reduce_mean(y_ * tf.log(tf.clip_by_value(y, 1e-10, 1.0)))
train_step = tf.train.AdamOptimizer(0.001).minimize(cross_entropy)
rdm = RandomState(1)
dataset_size = 128
X = rdm.rand(dataset_size, 2)

#Y = [[int(x1 + x2 < 1)] for (x1, x2) in X]
Y= [[int(x1 * x1 * x1 + x2 * x2 * x2 < 100)] for (x1, x2) in X]

with tf.Session() as sess:
    init_op = tf.global_variables_initializer()
    sess.run(init_op)
    print(sess.run(w1))
    print(sess.run(w2))

    STEPS = 20000
    for i in range(STEPS):
        start = (i * batch_size) % dataset_size
        end = min(start + batch_size, dataset_size)

        sess.run(train_step, feed_dict={x: X[start:end], y_: Y[start:end]})
        if i % 1000 == 0:
            total_cross_entropy = sess.run(cross_entropy, feed_dict={x: X, y_: Y})
            print("After %d training step(s), cross entropy on all data is %g" % (i, total_cross_entropy))

    print(sess.run(w1))
    print(sess.run(w2))

[[-0.81131822  1.48459876  0.06532937]
 [-2.4427042   0.0992484   0.59122431]]
[[-0.81131822]
 [ 1.48459876]
 [ 0.06532937]]
After 0 training step(s), cross entropy on all data is 0.470155
After 1000 training step(s), cross entropy on all data is 0.054546
After 2000 training step(s), cross entropy on all data is 0.0293009
After 3000 training step(s), cross entropy on all data is 0.0153037
After 4000 training step(s), cross entropy on all data is 0.0100591
After 5000 training step(s), cross entropy on all data is 0.00823807
After 6000 training step(s), cross entropy on all data is 0.00741278
After 7000 training step(s), cross entropy on all data is 0.0063658
After 8000 training step(s), cross entropy on all data is 0.00516446
After 9000 training step(s), cross entropy on all data is 0.00390483
After 10000 training step(s), cross entropy on all data is 0.00265029
After 11000 training step(s), cross entropy on all data is 0.00144403
After 12000 training step(s), cross entropy on all data 

#### Using activation function relu:
[[-0.81131822  1.48459876  0.06532937]
 [-2.4427042   0.0992484   0.59122431]]
[[-0.81131822]
 [ 1.48459876]
 [ 0.06532937]]
After 0 training step(s), cross entropy on all data is 0.470155
After 1000 training step(s), cross entropy on all data is 0.054546
After 2000 training step(s), cross entropy on all data is 0.0293009
After 3000 training step(s), cross entropy on all data is 0.0153037
After 4000 training step(s), cross entropy on all data is 0.0100591
After 5000 training step(s), cross entropy on all data is 0.00823807
After 6000 training step(s), cross entropy on all data is 0.00741278
After 7000 training step(s), cross entropy on all data is 0.0063658
After 8000 training step(s), cross entropy on all data is 0.00516446
After 9000 training step(s), cross entropy on all data is 0.00390483
After 10000 training step(s), cross entropy on all data is 0.00265029
After 11000 training step(s), cross entropy on all data is 0.00144403
After 12000 training step(s), cross entropy on all data is 0.00029683
After 13000 training step(s), cross entropy on all data is -0
After 14000 training step(s), cross entropy on all data is -0
After 15000 training step(s), cross entropy on all data is -0
After 16000 training step(s), cross entropy on all data is -0
After 17000 training step(s), cross entropy on all data is -0
After 18000 training step(s), cross entropy on all data is -0
After 19000 training step(s), cross entropy on all data is -0
[[-0.81131822  4.11521339  3.71558714]
 [-2.4427042   1.90406477  3.5929153 ]]
[[-0.81131822]
 [ 4.19540119]
 [ 2.67088747]]

#### Without using activation function
After 0 training step(s), cross entropy on all data is 0.0674925
After 1000 training step(s), cross entropy on all data is 0.0163385
After 2000 training step(s), cross entropy on all data is 0.00907547
After 3000 training step(s), cross entropy on all data is 0.00714436
After 4000 training step(s), cross entropy on all data is 0.00578471
After 5000 training step(s), cross entropy on all data is 0.00430222
After 6000 training step(s), cross entropy on all data is 0.00280812
After 7000 training step(s), cross entropy on all data is 0.00137464
After 8000 training step(s), cross entropy on all data is 2.11566e-05
After 9000 training step(s), cross entropy on all data is -0
After 10000 training step(s), cross entropy on all data is -0
After 11000 training step(s), cross entropy on all data is -0
After 12000 training step(s), cross entropy on all data is -0
After 13000 training step(s), cross entropy on all data is -0
After 14000 training step(s), cross entropy on all data is -0
After 15000 training step(s), cross entropy on all data is -0
After 16000 training step(s), cross entropy on all data is -0
After 17000 training step(s), cross entropy on all data is -0
After 18000 training step(s), cross entropy on all data is -0
After 19000 training step(s), cross entropy on all data is -0
[[-2.59392238  3.18602753  2.38825655]
 [-4.1101799   1.6826365   2.83427358]]
[[-2.43003726]
 [ 3.33411145]
 [ 2.10067439]]